command/e2e/register_validator.go

```diff
@@ -294,22 +294,22 @@ type txnSender struct {
 	account *wallet.Account
 }
 
-func (t *txnSender) sendTransaction(txn *types.Transaction) asyncTxn {
-	if txn.GasPrice == nil {
-		txn.GasPrice = big.NewInt(defaultGasPrice)
+func (t *txnSender) sendTransaction(txnData *types.LegacyTx) asyncTxn {
```

Comment:
Didn't want to have `*types.Transaction` here as a parameter?

Reply:
Not really because it is used within this package only and adds some extra fields population within the function itself. It would not be possible if we pass the common transaction model.